ftd/src/executor/styles.rs

```diff
@@ -823,22 +1032,365 @@ impl Background {
         }
     }
 
-    pub fn to_image_css_string(&self) -> String {
+    pub fn to_image_src_css_string(&self) -> String {
+        match self {
+            Background::Solid(_) => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
+            Background::Image(i) => i.to_image_src_css_string(),
+        }
+    }
+
+    pub fn to_image_repeat_css_string(&self) -> String {
         match self {
             Background::Solid(_) => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
-            Background::Image(i) => format!("url({})", i.light.value),
+            Background::Image(i) => i.to_repeat_css_string(),
+        }
+    }
+
+    pub fn to_image_size_css_string(&self) -> String {
+        match self {
+            Background::Solid(_) => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
+            Background::Image(i) => i.to_size_css_string(),
+        }
+    }
+
+    pub fn to_image_position_css_string(&self) -> String {
+        match self {
+            Background::Solid(_) => ftd::interpreter2::FTD_IGNORE_KEY.to_string(),
+            Background::Image(i) => i.to_position_css_string(),
         }
     }
 
     pub fn to_css_string(&self) -> String {
         match self {
             Background::Solid(c) => c.light.value.to_css_string(),
-            Background::Image(i) => i.light.value.to_string(),
+            Background::Image(i) => i.to_image_src_css_string(),
         }
     }
 
     pub fn background_image_pattern() -> (String, bool) {
-        ("url({0})".to_string(), false)
+        (
+            indoc::indoc! {"
+                let bg = {{0}};
+                if (typeof bg === 'object' && \"src\" in bg) {{
+                    let img_src = bg.src;
+                    if(!data[\"ftd#dark-mode\"] && typeof img_src === 'object' && \"light\" in img_src) {{
+                        \"url(\" + img_src.light + \")\"
+                    }}
+                    else if(data[\"ftd#dark-mode\"] && typeof img_src === 'object' && \"dark\" in img_src){{
+                        \"url(\" + img_src.dark + \")\"
+                    }}
+                    else {{
+                        null
+                    }}
+                }} else {{
+                    null
+                }}
+            "}.to_string(),
+            true,
+        )
+    }
+
+    pub fn background_repeat_pattern() -> (String, bool) {
+        (
+            format!(
+                indoc::indoc! {"
+                let bg = {{0}};
+                if (typeof bg === 'object' && \"repeat\" in bg) {{
+                    bg.repeat
+                }} else {{
+                    \"{ignore_key}\"
+                }}
+            "},
```

Comment:
```suggestion
                indoc::indoc! {r#"
                  let bg = {{0}};
                  if (typeof bg === 'object' && "repeat" in bg) {{
                      bg.repeat
                  }} else {{
                      "{}"
                  }}
              "#},
```

We should use a raw string (`r#""#`) for the pattern string to avoid the need to escape quotes.